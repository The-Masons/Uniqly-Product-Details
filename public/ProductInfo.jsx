import React from 'react';

class ProductInfo extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      headerShowing: 1
    }
    this.switchHeader = this.switchHeader.bind(this);
    this.handleHeaderClick = this.handleHeaderClick.bind(this);
  }

  switchHeader() {
    let newHeader;
    if (this.state.headerShowing === 1) {
      newHeader = 2;
    } else {
      newHeader = 1;
    }

    this.setState({
      headerShowing: newHeader
    })
  }

  handleHeaderClick(headerClicked) {
    if (this.state.headerShowing === headerClicked) {
      return;
    }

    this.switchHeader();
  }

  render() {
    //decide which header should be selected & which content to render
    let h1Class = 'detail-header1';
    let h2Class = 'detail-header2';
    let descClass = 'description';
    let matClass = 'materials';
    if (this.state.headerShowing === 1) {
      h1Class += ' detail-header-selected';
      matClass += ' hidden';
    } else {
      h2Class += ' detail-header-selected'
      descClass += ' hidden';
    }
    return (
      <div className="product-info">
        <div className={h1Class} onClick={() => this.handleHeaderClick(1)}>PRODUCT DETAIL</div>
        <div className={h2Class} onClick={() => this.handleHeaderClick(2)}>MATERIALS & CARE</div>
        <div className="tab-content">
          <div className={descClass}>
            {this.props.description}
          </div>
          <div className={matClass}>
            {this.props.materials}
          </div>
        </div>
      </div>
    );
  }
}

export default ProductInfo;