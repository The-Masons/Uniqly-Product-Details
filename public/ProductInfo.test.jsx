import React from 'react';
import ProductInfo from './ProductInfo.jsx';

const description = "It's good and you can wear it.";

const material = "This is the material for the product";

describe('ProductInfo', () => {
  test('should render with the correct items', () => {
    const productInfo = mount(
      <ProductInfo
        description={description} 
        materials={material}
        />
      );
    expect(productInfo).toMatchSnapshot();
  });

test('should correctly display the header', () => {
  const productInfo = mount(<ProductInfo />);

  productInfo.instance().switchHeader();

  expect(productInfo.state('headerShowing')).toEqual(2);
});

test('should not switch headers if the header is already showing', () => {
  const productInfo = mount(<ProductInfo />);

  productInfo.instance().handleHeaderClick(1);
  productInfo.instance().handleHeaderClick(1);

  expect(productInfo.state('headerShowing')).toEqual(1);

  productInfo.instance().handleHeaderClick(1);

  expect(productInfo.state('headerShowing')).toEqual(1);
});

});