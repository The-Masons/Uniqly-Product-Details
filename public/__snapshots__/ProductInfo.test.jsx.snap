// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfo should render with the correct items 1`] = `
<ProductInfo
  description="It's good and you can wear it."
  materials="This is the material for the product"
>
  <div
    className="product-info"
  >
    <div
      className="detail-header1 detail-header-selected"
      onClick={[Function]}
    >
      PRODUCT DETAIL
    </div>
    <div
      className="detail-header2"
      onClick={[Function]}
    >
      MATERIALS & CARE
    </div>
    <div
      className="tab-content"
    >
      <div
        className="description"
      >
        It's good and you can wear it.
      </div>
      <div
        className="materials hidden"
      >
        This is the material for the product
      </div>
    </div>
  </div>
</ProductInfo>
`;
